distributed/tests/test_utils.py

```diff
@@ -635,3 +637,33 @@ def foo():
 
     with pytest.warns(DeprecationWarning, match="removed in version 1.2.3"):
         assert foo() == "bar"
+
+
+def test__clean_filename():
```

Comment:
Instead of testing for an exact match for the output of `_clean_filename`, we could try making slightly looser `assert` statements (e.g. `assert sys.exec_prefix not in result`) which still provide solid test coverage, but are less platform-specific and probably more robust. Additionally, you may find [`os.path.join`](https://docs.python.org/3/library/os.path.html#os.path.join) useful for this test.

For example, something along the lines of:

```diff
diff --git a/distributed/tests/test_utils.py b/distributed/tests/test_utils.py
index 8d1f1e9d..3c1fcc6b 100644
--- a/distributed/tests/test_utils.py
+++ b/distributed/tests/test_utils.py
@@ -642,28 +642,18 @@ def test_deprecated():
 def test__clean_filename():
     # python standard library case
     # path with sys.exec_prefix and no site-packages
-    path_no_sp = os.path.join.__code__.co_filename
-    clean_path_no_sp = _clean_filename(path_no_sp)
-
-    if f"lib{os.sep}" in path_no_sp:
-        assert (
-            clean_path_no_sp
-            == f"...{os.sep}lib{os.sep}python{sys.version_info[0]}.{sys.version_info[1]}{os.sep}posixpath.py"
-        )
-    elif f"lib64{os.sep}" in path_no_sp:
-        assert (
-            clean_path_no_sp
-            == f"...{os.sep}lib64{os.sep}python{sys.version_info[0]}.{sys.version_info[1]}{os.sep}posixpath.py"
-        )
+    filename = os.path.join.__code__.co_filename
+    result = _clean_filename(filename)
+    assert sys.exec_prefix not in result
+    assert result.startswith(os.path.join("...", "lib"))
+    assert f"python{sys.version_info[0]}.{sys.version_info[1]}" in result

     # Third-party package - site-packages
-    path_third_party = toolz.merge.__code__.co_filename
-    clean_path_third_party = _clean_filename(path_third_party)
-    assert clean_path_third_party == f"...{os.sep}toolz{os.sep}dicttoolz.py"
-
-    # Other case where esername in path
-    path_usr = (
-        os.path.expanduser("~") + f"{os.sep}repository{os.sep}folder{os.sep}filename.py"
-    )
-    clean_path_usr = _clean_filename(path_usr)
-    assert clean_path_usr == f"...{os.sep}repository{os.sep}folder{os.sep}filename.py"
+    filename = toolz.merge.__code__.co_filename
+    result = _clean_filename(filename)
+    assert sys.exec_prefix not in result
+    assert result.startswith(os.path.join("...", "toolz"))
+
+    # Other case where username in path
+    filename = os.path.join(os.path.expanduser("~"), "dir", "filename.py")
+    assert _clean_filename(filename) == filename.replace(os.path.expanduser("~"), "...")
```